src/components/dialogs/parameters/widget/ParameterLine.tsx

```diff
@@ -0,0 +1,159 @@
+            );
+
+        case ParameterType.DropDown:
+            return (
+                <>
+                    <Grid item xs={8} className={classes.parameterName}>
+                        <FormattedMessage id={props.labelTitle} />
+                    </Grid>
+                    <Grid item container xs={4} className={classes.controlItem}>
+                        <Select
+                            labelId={props.labelValue}
+                            value={
+                                props.defaultValueIfNull && !parameterValue
+                                    ? Object.entries<string>(props.values)[0]
+                                    : parameterValue
+                            }
+                            onChange={(event) => {
+                                handleChangeParameterValue(event.target.value);
+                            }}
+                            size="small"
+                            disabled={props.disabled ?? false}
+                        >
+                            {Object.entries<string>(props.values).map(
+                                ([key, value]) => (
+                                    <MenuItem key={key} value={key}>
+                                        <FormattedMessage id={value} />
+                                    </MenuItem>
+                                )
+                            )}
+                        </Select>
+                    </Grid>
+                </>
+            );
+
+        case ParameterType.Slider:
+            // eslint-disable-next-line react-hooks/rules-of-hooks
+            const [sliderValue, setSliderValue] = useState(
+                Number(parameterValue)
+            );
+            return (
+                <>
+                    <Grid item xs={8} className={classes.parameterName}>
+                        <FormattedMessage id={props.label} />
+                    </Grid>
+                    <Grid item container xs={4} className={classes.controlItem}>
+                        <Slider
+                            min={props.minValue ?? 0}
+                            max={props.maxValue ?? 100}
+                            valueLabelDisplay="auto"
+                            onChange={(event, newValue) => {
+                                // @ts-ignore
+                                setSliderValue(newValue);
+                            }}
+                            onChangeCommitted={(event, value) => {
+                                handleChangeParameterValue(value);
+                            }}
+                            value={sliderValue}
+                            disabled={props.disabled ?? false}
+                            marks={props.marks}
+                        />
+                    </Grid>
+                </>
+            );
+
+        default:
+            //TODO: how to manage in react component? throw new Error(`Not supported parameter type ${type}`);
+            // @ts-ignore
```

Comment:
same here

Reply:
same, tslint warning

Reply:
the strange thing is that this warning was a bug, and previously fixed ...
* https://github.com/microsoft/TypeScript/issues/9838

solution found with a simple cast: https://github.com/Microsoft/TypeScript/issues/10600#issuecomment-243327310